### PR TITLE
Expanded navigation interfaces

### PR DESCRIPTION
**Pull request by abratashev**
UI support of https://github.com/tesler-platform/tesler/issues/18

### CODE REVIEW

**Comment by abratashev** [comment]
First commit:
https://github.com/tesler-platform/tesler-ui/pull/40

File: src/interfaces/navigation.ts [hunk 1]
```diff
@@ -1,64 +1,71 @@
...
- * TODO
- * @param item 
+ * Returns MenuItem if it is ViewNavigationGroup
+ *
+ * @param item to be identified as group
  */
-export function isViewNavigationGroup(item: any): item is ViewNavigationGroup {
-    return !!item && ('title' in item)
+export function isViewNavigationGroup(item: MenuItem): item is ViewNavigationGroup {
+    return !!item && ('child' in item)
```
Comment (Dergash) [inline src/interfaces/navigation.ts]: It's a breaking change, we should implement it gradually:
```tsx
    // TODO: remove 'title' check in 2.0.0
    return !!item && ('child' in item || 'title' in item)
```

File: src/interfaces/navigation.ts [hunk 2]
```diff
@@ -1,64 +1,71 @@
...
  *
- * @param title Имя группы.
- * @param child Список категорий или пунктов меню, входящих в группу.
+ * @param title Title of group. Navigation element shows it to user.
+ * @param child Array of navigation elements specified below group(View or inner Group)
  */
 export interface ViewNavigationGroup {
-    title: string
-    child: Array<ViewNavigationItem | ViewNavigationCategory>
+    /** TODO identifier will be nullable in next major */
```
Comment (Dergash) [inline src/interfaces/navigation.ts]: It'll be easier to find all planned TODOs if they'll mention target version directly:
```tsx
/** TODO identifier will be nullable in 2.0.0 */
```

File: src/interfaces/navigation.ts (hunk 1 above)
Comment (abratashev) [reply src/interfaces/navigation.ts]: Fixed.

File: src/interfaces/navigation.ts (hunk 2 above)
Comment (abratashev) [reply src/interfaces/navigation.ts]: Fixed.

**Pull request merged**